---
title: "02d_subsetforprojects.Rmd"
author: "Rayna M Harris"
date: "March 6, 2017"
output: md_document
---

Not all of my samples need to be processed together. This script will make the appropriate subset files. First, read in the new data

```{r install packages load data, warning=FALSE, message=FALSE}
library("dplyr") ## for filtering and selecting rows
library("plyr") ## for renaming factors
```


```{r dissociation}
DissociationCountData <- read.csv("../data/rnaseq/countbygene.csv", header=T, check.names = F, row.names = 1)
DissociationColData <- read.csv("../data/rnaseq/kallistosamples.csv", header=T)

DissociationColData <- DissociationColData %>%
  filter(Group %in% c("homecage", "control")) %>%
  filter(Genotype != "FMR1") %>%
  filter(Mouse == "15-100") %>%
  filter(!grepl("C|D|16-", Mouse))  
savecols <- as.character(DissociationColData$RNAseqID) #selects all good samples
savecols <- as.vector(savecols) # make it a vector
DissociationCountData <- DissociationCountData %>% select(one_of(savecols)) # keep good samples

# For GEO database
DissociationMetaData <- DissociationColData %>%
  select(RNAseqID, Mouse, Region, Method) 
DissociationMetaData$title <- as.factor(paste(DissociationMetaData$Mouse, DissociationMetaData$Region, DissociationMetaData$Method,sep=" "))
DissociationMetaData$Organism <- "Mus musculus"
DissociationMetaData$molecule <- "RNA"
DissociationMetaData$file <- "/abuntance.txt"
DissociationMetaData$alignment <- as.factor(paste(DissociationMetaData$RNAseqID,DissociationMetaData$file, sep=""))
DissociationMetaData$strain <- "C57BL/6"

#write.csv(DissociationMetaData, "~/Github/DissociationTest/data/DissociationMetaData.csv", row.names=T)

## rename, relevel, slim, and prep for DESEq2 
rownames(DissociationColData) <- DissociationColData$RNAseqID
DissociationColData <- DissociationColData %>%
  filter(Mouse %in% c("15-100")) %>% droplevels()
savecols <- as.character(DissociationColData$RNAseqID) #selects all good samples
savecols <- as.vector(savecols) # make it a vector
DissociationCountData <- DissociationCountData %>% select(one_of(savecols)) # keep good samples
DissociationColData <- rename(DissociationColData, c("Method"="Treatment"))
colnames(DissociationColData)
DissociationColData <- DissociationColData %>%
  select(RNAseqID,Mouse, Genotype,jobnumber, Region, Treatment, Date)
write.csv(DissociationColData, "../data/DissociationColData.csv",row.names=F)

# save files to dissociation project directory
#write.csv(DissociationColData, "~/Github/DissociationTest/data/DissociationColData.csv", row.names=F)
#write.csv(DissociationCountData, "~/Github/DissociationTest/data/DissociationCountData.csv", row.names=T)

```


```{r fmr1}
fmr1CountData <- read.csv("../data/rnaseq/countbygene.csv", header=T, check.names = F, row.names = 1)
fmr1ColData <- read.csv("../data/rnaseq/kallistosamples.csv", header=T)

fmr1ColData <- fmr1ColData %>%
  filter(jobnumber == "JA17009")
savecols <- as.character(fmr1ColData$RNAseqID) #selects all good samples
savecols <- as.vector(savecols) # make it a vector
fmr1CountData <- fmr1CountData %>% select(one_of(savecols)) # keep good samples

#save informative columns
fmr1ColData <- fmr1ColData %>%
  select(RNAseqID, Mouse, Genotype, Region, APA, Date, daytime)

# make meta data
fmr1MetaData <- fmr1ColData %>%
  select(RNAseqID, Mouse, Region, Genotype) 
fmr1MetaData$title <- as.factor(paste(fmr1MetaData$Mouse, fmr1MetaData$Region, fmr1MetaData$Genotype,sep=" "))
fmr1MetaData$Organism <- "Mus musculus"
fmr1MetaData$molecule <- "RNA"
fmr1MetaData$file <- "/abuntance.txt"
fmr1MetaData$alignment <- as.factor(paste(fmr1MetaData$RNAseqID,fmr1MetaData$file, sep=""))
names(fmr1MetaData)[4] <- "strain"
fmr1MetaData$strain <- ifelse(grepl("WT", fmr1MetaData$strain), "C57BL/6", "Fmr1-KO")

# save files to dissociation project directory
write.csv(fmr1ColData, "../data/rnaseq/fmr1ColData.csv", row.names=F)
write.csv(fmr1CountData, "../data/rnaseq/fmr1CountData.csv", row.names=T)
write.csv(fmr1MetaData, "../data/rnaseq/fmr1MetaData", row.names=F)

```






```{r WT2015withHomeCage}
## WithHomecage
IntegrativeWT2015countData <- read.csv("../data/rnaseq/countbygene.csv", header=T, check.names = F, row.names = 1)
IntegrativeWT2015colData <- read.csv("../data/rnaseq/kallistosamples.csv", header=T)
IntegrativeWT2015colData <- IntegrativeWT2015colData %>%
  filter(!grepl("16-", Mouse)) %>%
  filter(!grepl("100", Mouse)) 
  
savecols <- as.character(IntegrativeWT2015colData$RNAseqID) #selects all good samples
savecols <- as.vector(savecols) # make it a vector
IntegrativeWT2015countData <- IntegrativeWT2015countData %>% select(one_of(savecols)) # keep good samples

IntegrativeWT2015colData <- IntegrativeWT2015colData %>%
  select(RNAseqID, Mouse, year, Genotype, Region, jobnumber, Group, APA, Conflict)

IntegrativeWT2015colData$APA_Conflict <- as.factor(paste(IntegrativeWT2015colData$APA, IntegrativeWT2015colData$Conflict, sep="_"))

IntegrativeWT2015colData$Treatment <- ifelse(grepl("NA_NA", IntegrativeWT2015colData$APA_Conflict), "homecage", 
                                    ifelse(grepl("Trained_Conflict", IntegrativeWT2015colData$APA_Conflict), "conflict",
                                    ifelse(grepl("Trained_NoConflict", IntegrativeWT2015colData$APA_Conflict), "trained",
                                    ifelse(grepl("Yoked_NoConflict", IntegrativeWT2015colData$APA_Conflict), "yoked",
                                    ifelse(grepl("Yoked_Conflict", IntegrativeWT2015colData$APA_Conflict), "shocked", "NA")))))
                                                                                                  
# save files to IntegrativeWT2015 project directory
write.csv(IntegrativeWT2015colData, "~/Github/IntegrativeProjectWT2015/data/IntegrativeWT2015ColData.csv", row.names=F)
write.csv(IntegrativeWT2015countData, "~/Github/IntegrativeProjectWT2015/data/IntegrativeWT2015CountData.csv", row.names=T)


# save files to Dissociation project directory
write.csv(IntegrativeWT2015colData, "~/Github/DissociationTest/data/StressCognitionTestColData.csv", row.names=F)
write.csv(IntegrativeWT2015countData, "~/Github/DissociationTest/data/StressCognitionTestCountData.csv", row.names=T)
```


## Session Info
```{r SessionInfo}
sessionInfo()
```