---
title: "02c_DESeq2"
author: "Rayna M Harris"
date: "March 6, 2017"
output: md_document
---

This will do a quick and dirty DESeq2 analysis for all samples!

```{r install packages load data, warning=FALSE, message=FALSE}
library("ggplot2") ## pretty plots
library("DESeq2")
library("genefilter") # for selecting top varying genes
library("pheatmap") # for the pretty heatmap
knitr::opts_chunk$set(fig.path = '../results/02c_DESeq2/') # set output file for figures 
```


```{r DESeq}
countData <- read.csv("../data/rnaseq/countbygene.csv", header=T, check.names = F, row.names = 1)
colData <- read.csv("../data/rnaseq/kallistosamples.csv", header=T)
rownames(colData) <- colData$RNAseqID

## differential gene expression
dds <- DESeqDataSetFromMatrix(countData = countData,
  colData = colData,
  design = ~ Group + Region)
dds

## 1.3.6 Pre-filtering
dds <- dds[ rowSums(counts(dds)) > 1, ]

## 1.3.7 Note on factor levels
dds$Group <- factor(dds$Group, levels=c("homecage","control","consistent", "conflict"))
dds$Region <- factor(dds$Region, levels=c("DG","CA1", "CA3"))

## 1.4  Differential expression analysi
dds <- DESeq(dds)

# general deseq
res <- results(dds, independentFiltering = F)
resOrdered <- res[order(res$padj),]
head(resOrdered,10)

sum(res$padj < 0.1, na.rm = TRUE) 
res05 <- results(dds, alpha=0.05)
table(res05$padj < .05)
sum(res05$padj < 0.05, na.rm=TRUE)


## 1.5 exploring and reporting results

plotMA(res, main="plotMA")

resMLE <- results(dds)
head(resMLE, 4)

hist(res$pvalue[res$baseMean > 1], breaks=0:20/20, col="grey50", border="white")

plotCounts(dds, gene=which.min(res$padj), intgroup="Group")
plotCounts(dds, gene=which.min(res$padj), intgroup="Punch")

respadj <- as.data.frame(res$padj)
head(respadj)

## 1.5 more info
mcols(res)$description

## for variance stablized gene expression and log transformed data
rld <- rlog(dds, blind=FALSE)
vsd <- varianceStabilizingTransformation(dds, blind=FALSE)
vsd.fast <- vst(dds, blind=FALSE)
head(assay(rld), 3)
```


## pca plot

```{r pca}
pcaData <- plotPCA(rld, intgroup = c( "Group", "Punch"), returnData=TRUE)
pcaData
percentVar <- round(100 * attr(pcaData, "percentVar"))

ggplot(pcaData, aes(PC1, PC2, color=Group, shape = Punch)) + geom_point(size=3) +
xlab(paste0("PC1: ",percentVar[1],"% variance")) +
ylab(paste0("PC2: ",percentVar[2],"% variance")) +
coord_fixed()
```

```{r heatmap}
topVarGenes <- head(order(rowVars(assay(rld)),decreasing=TRUE),25)
mat <- assay(rld)[ topVarGenes, ]
mat <- mat - rowMeans(mat)
df <- as.data.frame(colData(rld)[,c("Group", "Punch")])
pheatmap(mat, show_colnames=F, show_rownames = T,
annotation_col=df)
```


## Session Info
```{r SessionInfo}
sessionInfo()
```