---
title: "FMR1"
author: "Rayna M Harris"
date: "2/24/2017"
output: md_document
---

## Loading the data
In the summer of 2016, I processed a bunch of hippocampal tissue samples from WT and FMR1-KO mice that were trained in an active place avoidance task. 

This data was added the the epically large sample collection database contained in the two files  "animals.csv" and "punches.csv" which provided a detailed account of all animals processed and all tissue samples collected. Then, I tidy the dataframe a little bit to get it prepared for RNAseq sample submision.

```{r install packages load data, warning=FALSE, message=FALSE}
#install.packages("tidyr", dependencies=TRUE)
#source("https://bioconductor.org/biocLite.R")
#biocLite("DESeq2")
library("tidyr") 
library("dplyr") ## for filtering and selecting rows
library("plyr")  ## for renmaing factors
library("reshape2") ##  for melting dataframe
library("ggplot2") ## for awesome plots!
library("magrittr") ## to use the weird pipe
library("gplots") ##for making awesome plots
library("cowplot") ## for some easy to use themes
library("DESeq2") ## for differnetial gene expression profiling
library("ggrepel") ## for labeling volcano plot

# set output file for figures 
knitr::opts_chunk$set(fig.path = '../results/fmr1/')

source("functions_behavior.R")
source("figureoptions.R")
```

Read the data

```{r ImportData}
# this starts with data genearated from code described in 01a_behavdatawrangling.R 
behavior <- read.csv('../data/behavior/fmr1.csv' , header = T)

# this starts with data genearated from code described in 02a_tidysamples.R and 02b_KallistoGather.Rmd
colData <- read.csv('../data/rnaseq/fmr1ColData.csv')
rownames(colData) <- colData$RNAseqID
countData <-  read.csv('../data/rnaseq/fmr1CountData.csv', check.names = F, row.names = 1)

```


## Behavior analysis

```{r behaviordatawrangle}
## sert and revalue factors
behavior$APA <- factor(behavior$APA, levels = c("control", "consistent", "conflict"))
behavior$Genotype <- factor(behavior$Genotype, levels = c("WT", "FMR1KO"))
behavior$Time1stEntrLog <- log(behavior$Time1stEntr) 
summary(behavior)
behavior <- na.omit(behavior)

## behavior summary time
behaviorsummaryNum <- summarise(group_by(behavior, APA, Genotype, TrainSessionComboNum), m = mean(NumEntrances), se = sd(NumEntrances)/sqrt(length(NumEntrances)))
behaviorsummaryNum

firstentrance <- ggplot(behaviorsummaryNum, aes(x=, TrainSessionComboNum, y=m, color=APA, shape=Genotype)) + 
    geom_errorbar(aes(ymin=m-se, ymax=m+se, color=APA), width=.1) +
    geom_point(size = 4) +
   geom_line() +
    scale_y_continuous(name="Number of entrances") +
    scale_x_continuous(name = NULL, 
                       breaks = c(1, 2, 3, 4, 5, 6, 7, 8, 9),
                       labels=c("1" = "Hab.", "2" = "T1", "3" = "T2", 
                                "4" = "T3", "5" = "Retest", "6" = "T4/C1",
                                "7" = "T5/C2", "8" = "T6/C3", "9"= "Reten.")) +
  theme_cowplot(font_size = 14, line_size = 1) +
  background_grid(major = "none", minor = "none") +
  scale_color_manual(values = colorvalAPA) + 
  theme(legend.position="top") 
firstentrance



```





## RNAsequencing analysis

```{r removeoutliers}
## remove outliers
colData <- colData %>% filter(RNAseqID != "16-125B", RNAseqID != "16-123B") #set the coldata to be the countbygene df

## colData and countData must contain the exact same sample. I'll use the next three lines to make that happen
savecols <- as.character(colData$RNAseqID) #select the sample name column that corresponds to row names
savecols <- as.vector(savecols) # make it a vector
countData <- countData %>% dplyr::select(one_of(savecols)) # select just the columns that match the samples in colData

```

## DESeq Analysis

Now, I'll look for differential gene expression between the FMR1-KO and WT mice. This analysis was developed by reading the DESEq manual. In many place, I try to provide the chapter where these steps are described in more details.

```{r DESeq, echo=F}
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ Genotype)
dds <- dds[ rowSums(counts(dds)) > 1, ] ## Pre-filtering
dds$Genotype <- factor(dds$Genotype, levels=c("WT","FMR1")) #factor level
dds <- DESeq(dds) ## 1.4  Differential expression analysi

res <- results(dds, independentFiltering = F) #results
resOrdered <- res[order(res$padj),] # orderd by smallest pval

summary(resOrdered)
head(resOrdered,10)

rld <- rlog(dds, blind=FALSE)
vsd <- varianceStabilizingTransformation(dds, blind=FALSE)
vsd.fast <- vst(dds, blind=FALSE)
head(assay(rld), 3)
```

## Data viz
```{r plots, echo=F}
plotMA(res, ylim=c(-1.5,1))

hist(res$pvalue[res$baseMean > 1], breaks=0:20/20, col="grey50", border="white")

plotCounts(dds, gene=which.min(res$padj), intgroup="Genotype")

## for variance stablized gene expression and log transformed data

```


## pca plot

```{r pca}
pcaData <- plotPCA(rld, intgroup = c( "Genotype", "Conflict"), returnData=TRUE)
pcaData
percentVar <- round(100 * attr(pcaData, "percentVar"))

ggplot(pcaData, aes(PC1, PC2, color=Genotype, label = name)) + geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  geom_text() +
  coord_fixed()
```

```{r heatmap}
library("genefilter")
library("pheatmap")
topVarGenes <- head(order(rowVars(assay(rld)),decreasing=TRUE),25)
mat <- assay(rld)[ topVarGenes, ]
mat <- mat - rowMeans(mat)
df <- as.data.frame(colData(rld)[,c("Genotype")])
pheatmap(mat)



## differntiall expressed heatmap
source("resvalsfunction.R")
DEGes <- assay(rld)
contrast1 <- resvals(contrastvector = c("Genotype", "FMR1", "WT"), mypadj = 0.1)
DEGes <- cbind(DEGes, contrast1)
DEGes <- as.data.frame(DEGes) # convert matrix to dataframe
DEGes$rownames <- rownames(DEGes)  # add the rownames to the dataframe
DEGes$pvaljmin <- with(DEGes, pmin(pvalGenotypeFMR1WT)) # create new col with min pval
DEGes <- DEGes %>% filter(pvaljmin < 0.0001)
rownames(DEGes) <- DEGes$rownames
drop.cols <-colnames(DEGes[,grep("padj|pval|rownames", colnames(DEGes))])
DEGes <- DEGes %>% dplyr::select(-one_of(drop.cols))
DEGes <- as.matrix(DEGes)
DEGes <- DEGes - rowMeans(DEGes)
head(DEGes)

## set anntation variables
df <- as.data.frame(colData(dds)[,c("Genotype","Conflict")]) ## matrix to df

# set color breaks
paletteLength <- 30
myBreaks <- c(seq(min(DEGes), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(DEGes)/paletteLength, max(DEGes), length.out=floor(paletteLength/2)))

colorpalette <-  colorRampPalette(c("Deep Sky Blue 3", "white", "red"))( 30 )


pheatmap(DEGes, show_colnames=F, show_rownames = T,
         annotation_col=df, 
         #annotation_colors = ann_colors,
         treeheight_row = 0, treeheight_col = 25,
         #fontsize = 11, 
         #width=4.5, height=3,
         border_color = "grey60" ,
         color = colorpalette,
         #cellwidth = 12, 
         clustering_method="average",
         breaks=myBreaks,
         clustering_distance_cols="correlation" 
         )
```


## Volcano plot

```{r volcanoplot}
res <- results(dds, contrast =c("Genotype", "FMR1", "WT"), independentFiltering = F)
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot"))
with(subset(res, log2FoldChange>0), points(log2FoldChange, -log10(pvalue), pch=20, col="orange"))
with(subset(res, log2FoldChange<0), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
with(subset(res, padj>.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="grey"))


resOrdered <- res[order(res$padj),]
head(resOrdered)



data <- data.frame(gene = row.names(res), pvalue = -log10(res$padj), lfc = res$log2FoldChange)
data <- na.omit(data)
head(data)

data <- data %>%
  mutate(color = ifelse(data$lfc > 0 & data$pvalue > 1.3, 
                        yes = "WT", 
                        no = ifelse(data$lfc < 0 & data$pvalue > 1.3, 
                                    yes = "FMR1", 
                                    no = "none")))
top_labelled <- top_n(data, n = 9, wt = pvalue)

# Color corresponds to fold change directionality
colored <- ggplot(data, aes(x = lfc, y = pvalue)) + 
  geom_point(aes(color = factor(color)), size = 1.75, alpha = 0.8, na.rm = T) + # add gene points
  theme_bw(base_size = 16) + # clean up theme
  theme(legend.position = "none") + # remove legend 
  scale_color_manual(values = c("FMR1" = "#7570b3",
                                "WT" = "#d95f02", 
                                "none" = "#bdbdbd")) + theme(panel.grid.minor=element_blank(),
           panel.grid.major=element_blank()) + 
  theme(axis.title.x = element_blank())+ 
  theme(axis.title.y = element_blank()) + 
  geom_text_repel(data = top_labelled, 
                          mapping = aes(label = gene), 
                          size = 3,
                          fontface = 'bold', 
                          color = 'black',
                          box.padding = unit(0.5, "lines"),
                          point.padding = unit(0.5, "lines"))

colored



```

## Session Info
```{r session info}
sessionInfo()
```