---
title: "FMR1"
author: "Rayna M Harris"
date: "2/24/2017"
output: md_document
---

## Loading the data
In the summer of 2016, I processed a bunch of hippocampal tissue samples from WT and FMR1-KO mice that were trained in an active place avoidance task. 

This data was added the the epically large sample collection database contained in the two files  "animals.csv" and "punches.csv" which provided a detailed account of all animals processed and all tissue samples collected. Then, I tidy the dataframe a little bit to get it prepared for RNAseq sample submision.

```{r install packages load data, warning=FALSE, message=FALSE}
#install.packages("tidyr", dependencies=TRUE)
#source("https://bioconductor.org/biocLite.R")
#biocLite("DESeq2")
library("tidyr") 
library("dplyr") ## for filtering and selecting rows
library("plyr")  ## for renmaing factors
library("reshape2") ##  for melting dataframe
library("ggplot2") ## for awesome plots!
library("magrittr") ## to use the weird pipe
library("gplots") ##for making awesome plots
library("cowplot") ## for some easy to use themes
library("DESeq2") ## for differnetial gene expression profiling

# set output file for figures 
knitr::opts_chunk$set(fig.path = '../results/fmr1/')

```

Read the data



```{r ImportData, echo=FALSE, message=FALSE, comment=FALSE, warning=FALSE}
# this starts with data genearated from code described in ../../BehavEphysRNAseq/markdownfiles/02a_tidysamples.R and ../../BehavEphysRNAseq/markdownfiles/02b_KallistoGather.Rmd
colData <- read.csv('../data/rnaseq/fmr1ColData.csv')
rownames(colData) <- colData$RNAseqID
countData <-  read.csv('../data/rnaseq/fmr1CountData.csv', check.names = F, row.names = 1)
```



## DESeq Analysis

Now, I'll look for differential gene expression between the FMR1-KO and WT mice. This analysis was developed by reading the DESEq manual. In many place, I try to provide the chapter where these steps are described in more details.

```{r DESeq}

## remove genes with total counts across all samples < 2
countData[countData < 2] <- 0

## differential gene expression
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ Genotype)
dds


## 1.3.6 Pre-filtering
dds <- dds[ rowSums(counts(dds)) > 1, ]

## 1.3.7 Note on factor levels
dds$Genotype <- factor(dds$Genotype, levels=c("WT","FMR1"))

## 1.4  Differential expression analysi
dds <- DESeq(dds)

# general deseq
res <- results(dds, independentFiltering = F)
resOrdered <- res[order(res$padj),]
summary(res)
head(resOrdered,10)
sum(res$padj < 0.1, na.rm = TRUE) 
res05 <- results(dds, alpha=0.05)
summary(res05) 
table(res05$padj < .05)
sum(res05$padj < 0.05, na.rm=TRUE)


## 1.5 exploring and reporting results

plotMA(res, main="plotMA")

plotMA(res, ylim=c(-2,2))

resMLE <- results(dds)
head(resMLE, 4)

hist(res$pvalue[res$baseMean > 1], breaks=0:20/20, col="grey50", border="white")

plotCounts(dds, gene=which.min(res$padj), intgroup="Genotype")

respadj <- as.data.frame(res$padj)
head(respadj)

## 1.5 more info
mcols(res)$description

## for variance stablized gene expression and log transformed data
rld <- rlog(dds, blind=FALSE)
vsd <- varianceStabilizingTransformation(dds, blind=FALSE)
vsd.fast <- vst(dds, blind=FALSE)
head(assay(rld), 3)

```


## pca plot

```{r pca}
pcaData <- plotPCA(rld, intgroup = c( "Genotype"), returnData=TRUE)
pcaData
percentVar <- round(100 * attr(pcaData, "percentVar"))

ggplot(pcaData, aes(PC1, PC2, color=Genotype)) + geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  coord_fixed()
```

```{r heatmap}
library("genefilter")
library("pheatmap")
topVarGenes <- head(order(rowVars(assay(rld)),decreasing=TRUE),25)
mat <- assay(rld)[ topVarGenes, ]
mat <- mat - rowMeans(mat)
df <- as.data.frame(colData(rld)[,c("Genotype")])
pheatmap(mat)
```


## DESeq2 with transcripts, not genes!!!

```{r DESeqTranscriptsNotGenes}
# 1.3.3 Count matrix input ----
countData <- count 
countData <- countData %>% 
  mutate_each(funs(round(.,0)), -id) 
str(countData)
colData <- Traits %>%
  arrange(RNAseqID) # needs to be in same order a countData
head(countData)
head(colData)



## removing outliers. if the colDatat dataframe has fewer samples than the colData, then this will remove that sample from the analysis

colData <- colData %>%
  dplyr::filter(!grepl("16-125B|16-116D|16-123B", RNAseqID)) 

savecols <- as.character(colData$RNAseqID) #select the sample name column that corresponds to row names
savecols <- as.vector(savecols) # make it a vector
countData <- countData %>% select(one_of(savecols)) # select just the columns that match the samples in colData

## remove genes with total counts across all samples < 2
countData[countData < 2] <- 0

## differential gene expression
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData,
                              design = ~ Genotype)
dds


## 1.3.6 Pre-filtering
dds <- dds[ rowSums(counts(dds)) > 1, ]

## 1.3.7 Note on factor levels
dds$Genotype <- factor(dds$Genotype, levels=c("WT","FMR1"))

## 1.4  Differential expression analysi
dds <- DESeq(dds)

# general deseq
res <- results(dds, independentFiltering = F)
resOrdered <- res[order(res$padj),]
summary(res)
head(resOrdered,10)
sum(res$padj < 0.1, na.rm = TRUE) 
res05 <- results(dds, alpha=0.05)
summary(res05) 
table(res05$padj < .05)
sum(res05$padj < 0.05, na.rm=TRUE)


## 1.5 exploring and reporting results

plotMA(res, main="plotMA")

plotMA(res, ylim=c(-2,2))

resMLE <- results(dds)
head(resMLE, 4)

hist(res$pvalue[res$baseMean > 1], breaks=0:20/20, col="grey50", border="white")

plotCounts(dds, gene=which.min(res$padj), intgroup="Genotype")

respadj <- as.data.frame(res$padj)
head(respadj)

## 1.5 more info
mcols(res)$description

## for variance stablized gene expression and log transformed data
rld <- rlog(dds, blind=FALSE)
vsd <- varianceStabilizingTransformation(dds, blind=FALSE)
vsd.fast <- vst(dds, blind=FALSE)
head(assay(rld), 3)

pcaData <- plotPCA(rld, intgroup = c( "Genotype"), returnData=TRUE)
pcaData
percentVar <- round(100 * attr(pcaData, "percentVar"))

ggplot(pcaData, aes(PC1, PC2, color=Genotype)) + geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  coord_fixed()

library("genefilter")
library("pheatmap")
topVarGenes <- head(order(rowVars(assay(rld)),decreasing=TRUE),25)
mat <- assay(rld)[ topVarGenes, ]
mat <- mat - rowMeans(mat)
df <- as.data.frame(colData(rld)[,c("Genotype")])
pheatmap(mat)
```









## Session Info
```{r session info}
sessionInfo()
```